aws/src/main/java/org/apache/iceberg/aws/glue/HadoopConfigurableGlueCatalog.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.Map;
+import org.apache.hadoop.conf.Configurable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.aws.s3.S3FileIO;
+import org.apache.iceberg.io.FileIO;
+
+/**
+ * Glue catalog that also accepts Hadoop configuration,
+ * for users that would like to use GlueCatalog with HadoopFileIO or
+ * other dynamic FileIO implementation that leverages Hadoop configuration.
+ */
+public class HadoopConfigurableGlueCatalog extends GlueCatalog implements Configurable {
```

Comment:
Instead of splitting into two classes, what if we just updated the IO initialization to dynamically call `Configurable.setConf`? Then we could avoid this problem for all implementations of `FileIO`.